core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -437,6 +480,41 @@ private ExprValue exprAddDateDays(ExprValue date, ExprValue days) {
         : exprValue);
   }
 
+  /**
+   * CONVERT_TZ function implementation for ExprValue.
+   * Returns null for time zones outside of -13:59 and +14:00. Matches MySQL range.
+   *
+   * @param startingDateTime ExprValue of DateTime that is being converted from
+   * @param fromTz           ExprValue of time zone, representing the time to convert from.
+   * @param toTz             ExprValue of time zone, representing the time to convert to.
+   * @return DateTime that has been converted to the to_tz timezone.
+   */
+  private ExprValue exprConvertTZ(ExprValue startingDateTime, ExprValue fromTz, ExprValue toTz) {
+    if (startingDateTime.type() == ExprCoreType.STRING) {
+      startingDateTime = exprDateTimeNoTimezone(startingDateTime);
+    }
+    try {
+      ZoneId convertedFromTz = ZoneId.of(fromTz.stringValue());
+      ZoneId convertedToTz = ZoneId.of(toTz.stringValue());
+
+      // isTimeZoneValid checks if the timezone is within the range accepted by MySQL standard.
+      if (!isTimeZoneValid(convertedFromTz)
```

Comment:
Is this necessary, since we're also catching for invalid timezones?

Reply:
I think I was confused here.  The isTimeZoneValid is checking against the SQL language standard.  If we renamed this function, I would be less confused.